Xamarin.Forms.Core/DataTemplateExtensions.cs

```diff
@@ -7,16 +7,10 @@ public static class DataTemplateExtensions
 	{
 		public static DataTemplate SelectDataTemplate(this DataTemplate self, object item, BindableObject container)
 		{
-			var selector = self as DataTemplateSelector;
-			if (selector == null)
-				return self;
+			if (self is DataTemplateSelector selector)
+				return selector.SelectTemplate(item, container);
 
-			return selector.SelectTemplate(item, container);
-		}
-
-		public static object CreateContent(this DataTemplate self, object item, BindableObject container)
```

Comment:
Besides the slight refactor to use new code above, this member is now merged with `ElementTemplate` to provide an overload that can be used everywhere, regardless of the `DataTemplate` type.